clar2wasm/tests/wasm-generation/control_flow.rs

```diff
@@ -62,3 +96,22 @@ proptest! {
         );
     }
 }
+
+proptest! {
+    #![proptest_config(super::runtime_config())]
+
+    #[test]
+    fn begin((expr, expected_val, is_response_intermediary) in begin_strategy()) {
+        if is_response_intermediary{
+            crosscheck(
+                &expr,
+                Err(())
+            );
+        }else{
+            crosscheck(
+                &expr,
+                Ok(Some(expected_val.into()))
+            );
+        }
```

Comment:
Looks a bit weird to have a full `if-else` statement with the same body here. I would compute the expected part separately here and have a single call to crosscheck.

Reply:
Fixed